src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -2589,6 +2554,11 @@ public static DType Union(ref bool fError, DType type1, DType type2, bool useLeg
             // For Lazy Types, union operations must expand the current depth
             if (type1.IsLazyType)
             {
+                if (type1 == type2)
```

Comment:
>type1 == type2 [](http://example.com/codeflow?start=20&length=14)

What's the definition of equals for lazytype here? Is it purely referential?

Reply:
Implemented through the LazyTypeProvider